### PR TITLE
[KYUUBI #2070][FLINK] Support Flink job submission on yarn-session mode

### PR DESCRIPTION
**Pull request by deadwind4**
### _Why are the changes needed?_

Support submitting a job to Yarn. The 'flink-sql-client.jar' has supported submitting a job to Yarn. We only make a few changes.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by yaooqinn — APPROVED** [review]

File: externals/kyuubi-flink-sql-engine/bin/flink-sql-engine.sh [hunk 1]
```diff
@@ -37,7 +37,7 @@ FLINK_SQL_ENGINE_HOME="$(cd `dirname $0`/..; pwd)"
 if [[ "$FLINK_SQL_ENGINE_HOME" == "$KYUUBI_HOME/externals/engines/flink" ]]; then
   FLINK_SQL_ENGINE_LIB_DIR="$FLINK_SQL_ENGINE_HOME/lib"
   FLINK_SQL_ENGINE_JAR=$(find "$FLINK_SQL_ENGINE_LIB_DIR" -regex ".*/kyuubi-flink-sql-engine_.*\.jar")
-  FLINK_HADOOP_CLASSPATH="$INTERNAL_HADOOP_CLASSPATHS"
+  FLINK_HADOOP_CLASSPATH="$INTERNAL_HADOOP_CLASSPATHS:$HADOOP_CLASSPATH"
```
Comment (SteNicholas) [inline externals/kyuubi-flink-sql-engine/bin/flink-sql-engine.sh:40]: IMO, the `$INTERNAL_HADOOP_CLASSPATHS` includes the `$HADOOP_CLASSPATH`. Why do you add the `HADOOP_CLASSPATH`?

File: externals/kyuubi-flink-sql-engine/bin/flink-sql-engine.sh [hunk 2]
```diff
@@ -46,7 +46,7 @@ else
   FLINK_SQL_ENGINE_LIB_DIR="$FLINK_SQL_ENGINE_HOME/target"
   FLINK_SQL_ENGINE_JAR=$(find "$FLINK_SQL_ENGINE_LIB_DIR" -regex '.*/kyuubi-flink-sql-engine_.*\.jar$' | grep -v '\-javadoc.jar$' | grep -v '\-tests.jar$')
   _FLINK_SQL_ENGINE_HADOOP_CLIENT_JARS=$(find $FLINK_SQL_ENGINE_LIB_DIR -regex '.*/hadoop-client-.*\.jar$' | tr '\n' ':')
-  FLINK_HADOOP_CLASSPATH="${_FLINK_SQL_ENGINE_HADOOP_CLIENT_JARS%:}"
+  FLINK_HADOOP_CLASSPATH="${_FLINK_SQL_ENGINE_HADOOP_CLIENT_JARS%:}:$HADOOP_CLASSPATH"
```
Comment (SteNicholas) [inline externals/kyuubi-flink-sql-engine/bin/flink-sql-engine.sh:49]: Why not add `${HADOOP_CLASSPATH}:${HADOOP_CONF_DIR}:${YARN_CONF_DIR}`?

**Comment by SteNicholas** [comment]
@yaooqinn , which version would release the feature support?

File: externals/kyuubi-flink-sql-engine/bin/flink-sql-engine.sh (hunk 1 above)
Comment (SteNicholas) [reply externals/kyuubi-flink-sql-engine/bin/flink-sql-engine.sh:40]: You may refer to the #2082

Comment (SteNicholas) [reply externals/kyuubi-flink-sql-engine/bin/flink-sql-engine.sh:40]: You may refer to the #2082

**Comment by deadwind4** [comment]
I have run both development and distribution mode submit yarn-session and yarn-per-job Flink jobs.

**Comment by yanghua** [comment]
@SteNicholas After you have the result of verification, please share it. thanks.

**Review by SteNicholas — APPROVED** [review]
LGTM. @yanghua I. have verified the changes locally. Please help to merge.

**Comment by yaooqinn** [comment]
thanks @deadwind4 @SteNicholas @yanghua for the great work, merged to master/1.5

**Pull request closed without merging**